src/main/java/globalquake/ui/stationselect/StationSelectFrame.java

```diff
@@ -35,37 +75,77 @@ public void actionPerformed(ActionEvent actionEvent) {
             }
         });
 
-        add(stationSelectPanel, BorderLayout.CENTER);
-        add(createToolbar(), BorderLayout.PAGE_START);
+        JToolBar toolBar = createToolbar();
+        toggleButton.addActionListener(new AbstractAction() {
+            @Override
+            public void actionPerformed(ActionEvent actionEvent) {
+                if (toolBar.isVisible()) {
+                    toolBar.setVisible(false);
+                    toggleButton.setText(">");
+                } else {
+                    toolBar.setVisible(true);
+                    toggleButton.setText("<");
+                }
+            }
+        });
+        JPanel CenterPanel = new JPanel(new GridBagLayout());
+
+        GridBagConstraints gbc = new GridBagConstraints();
+        gbc.gridx = 0;
+        gbc.gridy = 0;
+        gbc.anchor = GridBagConstraints.LINE_START;
+        gbc.fill = GridBagConstraints.VERTICAL;
+        gbc.weighty = 1.0;
+
+        CenterPanel.add(togglePanel, gbc);
+
+        gbc.gridx = 0;
+        gbc.gridy = 0;
+        gbc.anchor = GridBagConstraints.CENTER;
+        gbc.fill = GridBagConstraints.BOTH;
+        gbc.weightx = 1.0;
+        gbc.weighty = 1.0;
+
+        CenterPanel.add(stationSelectPanel, gbc);
+
+        add(CenterPanel, BorderLayout.CENTER);
+        add(toolBar, BorderLayout.WEST);
         add(new StationCountPanel(databaseMonitorFrame, new GridLayout(1,4)), BorderLayout.SOUTH);
 
         pack();
         setLocationRelativeTo(databaseMonitorFrame);
         setResizable(true);
         setTitle("Select Stations");
 
-        java.util.Timer timer = new Timer();
-        timer.scheduleAtFixedRate(new TimerTask() {
-            public void run() {
+        javax.swing.Timer timer = new javax.swing.Timer(16, actionEvent -> {
+            if(stationRepaint){
                 stationSelectPanel.repaint();
             }
-        }, 0, 1000 / 40);
+            else{
+                toggleButton.repaint();
+            }
+        });
+        timer.start();
```

Comment:
I see that you were struggling with repainting that toggleButton and used a Timer to partially solve the issue. However, there is much better way of doing that - see the next comment